### DISCUSSION TITLE
Need to pass query params to endpoint using ProxyGetRequest

### DISCUSSION THREAD

**Discussion by fabrobles92** [paritytech/jsonrpsee#1485] [Q&A] [answered]
I need to pass a query params so I can do **localhost:5050/account_balance?contract_address=0x123** but dont know how to define a placeholder for this params when creating the endpoint, is this posible?:

`let middleware = tower::ServiceBuilder::new().option_layer(cors).layer(ProxyGetRequestLayer::new("/account_balance/:string", "dev_accountBalance")?).timeout(Duration::from_secs(20));`

**Comment by niklasad1** [comment, marked as answer]
Hey @fabrobles92 

It doesn't work because https://github.com/paritytech/jsonrpsee/blob/master/server/src/middleware/http/proxy_get_request.rs#L135 requires the path exactly match and it doesn't extract the query params from the URL and passes it on to the rpc call.

Thus, either open a PR to add support for that in ProxyGetRequest or just write your own middleware for that (you can use https://github.com/paritytech/jsonrpsee/blob/master/server/src/middleware/http/proxy_get_request.rs as template) for that
